### ISSUE TITLE
Easily shut down inactive notebooks

### ISSUE THREAD

**Issue by carstenbauer** [fonsp/Pluto.jl#1076]
When opening and closing (in the sense of closing the browser tab and/or going back to Pluto's main page) multiple notebooks in a Pluto session all previous notebooks are still active in the background. This can lead to unnecessary memory usage due to the still active Julia processes. To actually stop them from being active it seems like one has to click on all the closing "x" symbols on Pluto's main page.

Perhaps I've missed it but there doesn't seem to be a way to stop and close (in the sense of shutdown) a notebook in one step. Would be great to have a keyboard combination and/or a button/link for this.

(Jupyter has this as "Close and Halt".)

**Comment by fonsp** [comment]
Instead of a direct [feature request](https://github.com/fonsp/Pluto.jl/issues/182), can you rephrase your issue as a _statement of a problem_?

**Comment by crstnbr** [comment]
> Instead of a direct [feature request](https://github.com/fonsp/Pluto.jl/issues/182), can you edit your post, rephrasing your issue as a _statement of a problem_?

Done.

> Note that mirroring Jupyter features is not a design goal of Pluto.

Never thought it was.

**Comment by fonsp** [comment]
Thanks for editing your comment!

You mention the memory problem (which is valid, I also have more notebooks running than I intended), but I do not see how the second button fixes it. I do understand that an extra button would be convenient, but my concerns are:
- it might make our design look more crowded. The look we are going for is "minimal", "unintimidating" and "like a web article".
- this button would need to be created and _maintained_

I think the problem you mentioned (notebook are still running in the background) is caused by me _forgetting_ about a notebook, and moving on to something else. In that case, the extra button would not help. Some other solutions would be:
- Shut down inactive notebooks automatically after a timeout
- Detect that memory is running out and give a review list of least active notebooks that could be shut down

**Comment by crstnbr** [comment]
> I do understand that an extra button would be convenient, but my concerns are:
> 
> * it might make our design look more crowded. The look we are going for is "minimal", "unintimidating" and "like a web article".
> * this button would need to be created and _maintained_

Valid points. Perhaps a keyboard shortcut would suffice? It wouldn't touch the design at all and perhaps this feature is for the "advanced user" anyways? (In Jupyter, I also only every use the keyboard shortcut.)

> * Shut down inactive notebooks automatically after a timeout

That's a good idea, given that we can distinguish inactive notebooks running in the background from idle notebooks that I still have open in browser tabs.

> * Detect that memory is running out and give a review list of least active notebooks that could be shut down

Might be hard to get right, but perhaps worth a shot.

**Comment by fonsp** [comment]
> that I still have open in browser tabs.

I didn't consider that we could also check whether it is still open in a browser tab, nice! That, in addition with "no edits in the last X hours" could be the way to check for this

**Comment by pankgeorg** [comment]
I think this one falls under the class of problems that are not very well address, about **what happens _outside_ of the notebook editor**: 
1. How do we manage files
2. How do we manage running processes
3. How do we handle multiple workers at once/resources

Maybe that's more of a platform issue, which we don't address at the moment (because we can't do everything simultaneously)

**Comment by fonsp** [comment]
Continued in #2236
